src/chain_parsers/visualsign-solana/TESTING.md

```diff
@@ -0,0 +1,504 @@
+# Testing Guide for visualsign-solana
+
+This document establishes the testing philosophy and practices for all program presets in the visualsign-solana crate.
+
+## Test Philosophy
+
+### Unit Tests for Instruction Parsing
+
+All program preset tests (SPL Token, System Program, Stake Pool, etc.) follow the same principles:
+
+- **Each fixture tests ONE specific instruction** in isolation
+- **These are UNIT tests**, not integration tests for full multi-instruction transactions
+- **Fixtures are the source of truth** - they contain real transaction data from actual on-chain transactions
+- **No network dependencies** - test fixtures are manually extracted and committed to git
+- **Field names match explorer output** - makes manual verification against Solscan/explorers easy
+
+### Integration Testing
+
+For testing full multi-instruction transactions and end-to-end workflows, use higher-level integration tests in the parent visualsign-parser project, not at this crate level.
+
+## Creating Test Fixtures
+
+### Overview
+
+Test fixtures validate that our instruction parsers correctly extract and display parameters from real Solana transactions. Each fixture:
+
+1. References a real transaction from an explorer (Solscan, Solana Explorer, etc.)
+2. Contains the raw instruction data for a specific instruction within that transaction
+3. Specifies expected field values to validate parser output
+
+### Step-by-Step Guide
+
+#### Step 1: Find a Transaction
+
+Find a transaction on an explorer that contains the instruction type you want to test:
+- Solscan: `https://solscan.io/tx/<SIGNATURE>?cluster=<CLUSTER>`
+- Solana Explorer: `https://explorer.solana.com/tx/<SIGNATURE>?cluster=<CLUSTER>`
+
+**Note**: Most transactions contain multiple instructions. You'll extract just the one instruction you want to test.
+
+#### Step 2: Fetch Raw Transaction Data
+
+Use Solana RPC to get the raw transaction data:
+
+```bash
+# For devnet
+curl -X POST https://api.devnet.solana.com \
+  -H "Content-Type: application/json" \
+  -d '{
+    "jsonrpc":"2.0",
+    "id":1,
+    "method":"getTransaction",
+    "params":[
+      "<TRANSACTION_SIGNATURE>",
+      {
+        "encoding":"json",
+        "maxSupportedTransactionVersion":0
+      }
+    ]
+  }' | python3 -m json.tool > transaction.json
```

Comment:
What does json.tool do? Pretty print? I usually use jq, but I guess it's more common to have python3 available than jq?